File: 2020-10-30-client-libraries/ruby/demo_requests.rb
require 'stripe'
Stripe.api_key = 'sk_test...'

puts "Making requests!"

# Create a customer with no params
# customer = Stripe::Customer.create
# p customer

# Retrieve customer
# customer = Stripe::Customer.retrieve(
#   "cus_IVOWFjzG7w1jh4"
# )
# p customer
#

# Create customer with scalar params
# customer = Stripe::Customer.create(
#   email: 'Jenny.Rosen@example.com',
#   name: 'Jenny Rosen',
# )
# # email=Jenny.Rosen@example.com&name=Jenny%20Rosen
# p customer

# Create customer with enum
# customer = Stripe::Customer.create(
#   tax_exempt: 'exempt'
# )
# p customer

# Create customer with nested hash
# customer = Stripe::Customer.create(
#   payment_method: 'pm_card_visa',
#   invoice_settings: {
#     default_payment_method: 'pm_card_visa',
#   }
# )
# p customer

# Create customer with list of strings
# customer = Stripe::Customer.create(
#   preferred_locales: ['en', 'es']
# )
# p customer
# p customer.preferred_locales

# Update a customer's email address
# customer = Stripe::Customer.update(
#   "cus_IVOWFjzG7w1jh4",
#   email: 'jr2@example.com',
# )
# p customer.id
# p customer.email

# Update with nested hash
# customer = Stripe::Customer.update(
#   "cus_IVOWFjzG7w1jh4",
#   invoice_settings: {
#     custom_fields: [{
#       name: 'VAT',
#       value: '123ABC'
#     }]
#   }
# )
# p customer.invoice_settings

# Fetch a list of customers
# customers = Stripe::Customer.list
# p customers
# puts customers.data.map(&:id)

# Filter list of customers by email
# jenny_rosen_customers = Stripe::Customer.list(
#   email: 'jenny.rosen@example.com'
# )
# # /v1/customers?email=jenny.rosen@example.com
# p jenny_rosen_customers.data.map(&:id)
# p jenny_rosen_customers.data.map(&:email)

# Delete a customer
# customer = Stripe::Customer.delete(
#   "cus_IVOWFjzG7w1jh4"
# )
# p customer

# Create payment intent
# payment_intent = Stripe::PaymentIntent.create(
#   amount: 1000,
#   currency: 'usd',
# )
# p payment_intent.id

# Confirm a payment intent
# payment_intent = Stripe::PaymentIntent.confirm(
#   "pi_1HuORgCZ6qsJgndJxKl4gfnp",
#   payment_method: 'pm_card_visa'
# )
# p payment_intent
# p payment_intent.status
# First Fetch API call to retrieve an invoice
# invoice = Stripe::Invoice.list(limit: 1).data.first
# Second API call to retrieve the list of
# lines from the invoice.
# lines = invoice.lines.list(limit: 5)
# p lines
# Results in a request to path with the ID
# of the parent resource in the path:
#   /v1/invoices/in_xxx/lines

customer = Stripe::Customer.create({
  email: "jenny.rosen@example.com"
}, {
  stripe_account: 'acct_1Ey3h1BqeQ4DKpna'
})
p customer

File: 2020-10-30-client-libraries/python/demo_requests.py
import stripe
stripe.api_key = 'sk_test...'

print('Making requests!')

# Create a customer with no params
# customer = stripe.Customer.create()
# print(customer)

# Retrieve a customer
# customer = stripe.Customer.retrieve(
#     'cus_IVPkw3KcFOEz5G'
# )
# # /v1/customers/cus_IVPkw3KcFOEz5G
# print(customer)

# Create a customer with scalar values
# customer = stripe.Customer.create(
#     email='jenny.rosen@example.com',
#     name='Jenny Rosen',
# )
# # email=jenny.rosen@example.com&name=Jenny%20Rosen
# print(customer.name)
# print(customer.email)

# Create customer with enum
# customer = stripe.Customer.create(
#     tax_exempt='reverse'
# )
# print(customer.tax_exempt)

# Create customer with nested dict
# customer = stripe.Customer.create(
#     payment_method='pm_card_visa',
#     invoice_settings={
#         'default_payment_method': 'pm_card_visa',
#     }
# )
# print(customer.invoice_settings)

# Create a customer with list of strings
# customer = stripe.Customer.create(
#     preferred_locales=['en', 'es']
# )
# print(customer)
# print(customer.preferred_locales)

# Update a customer's email address
# customer = stripe.Customer.modify(
#     'cus_IVPutJCp28aWWh',
#     email='jr2@example.com'
# )
# print(customer.id)
# print(customer.email)

# Update customer with nested dict
# customer = stripe.Customer.modify(
#     'cus_IVPutJCp28aWWh',
#     invoice_settings={
#         'custom_fields': [{
#             'name': 'VAT',
#             'value': '123ABC'
#         }]
#     }
# )
# print(customer.invoice_settings)

# Fetch a list of customers
# customers = stripe.Customer.list()
# print(customers)
# print([cus.id for cus in customers.data])

# Filter list of customers by email
# customers = stripe.Customer.list(
#     email='jenny.rosen@example.com'
# )
# print(customers)
# print([cus.id for cus in customers.data])
# print([cus.email for cus in customers.data])

# Delete a customer
# customer = stripe.Customer.delete(
#     'cus_IVPkw3KcFOEz5G',
# )
# print(customer)

# Create a payment intent to confirm
# payment_intent = stripe.PaymentIntent.create(
#     amount=1000,
#     currency='usd'
# )
# print(payment_intent.id)
# print(payment_intent.status)

# Confirm a payment intent
# payment_intent = stripe.PaymentIntent.confirm(
#     'pi_1HuPDJCZ6qsJgndJJOtfVz0X',
#     payment_method='pm_card_visa',
# )
# print(payment_intent.id)
# print(payment_intent.status)

# First, fetch an invoice
# invoice = stripe.Invoice.list(limit=1).data[0]
#
# # Second, fetch lines for invoice through
# # the `lines` nested service method.
# lines = invoice.lines.list(limit=5)
# # /v1/invoices/in_xxx/lines
# print(lines)

# Create a customer on a connected account
customer = stripe.Customer.create(
    email='jenny.rosen@example.com',
    stripe_account='acct_1Ey3h1BqeQ4DKpna'
)
print(customer)
